packages/next/client/dev/dev-server-watcher.js

```diff
@@ -0,0 +1,158 @@
+import { getEventSourceWrapper } from './error-overlay/eventsource'
+
+export default function initializeDevServerWatcher() {
+  const shadowHost = document.createElement('div')
+  shadowHost.id = '__next-dev-server-watcher'
+  // Make sure container is fixed and on a high zIndex so it shows
+  shadowHost.style.position = 'fixed'
+  shadowHost.style.top = '0'
+  shadowHost.style.left = '0'
+  shadowHost.style.width = '100%'
+  shadowHost.style.zIndex = 99999
+  document.body.appendChild(shadowHost)
+
+  let shadowRoot
+  let prefix = ''
+
+  if (shadowHost.attachShadow) {
+    shadowRoot = shadowHost.attachShadow({ mode: 'open' })
+  } else {
+    // If attachShadow is undefined then the browser does not support
+    // the Shadow DOM, we need to prefix all the names so there
+    // will be no conflicts
+    shadowRoot = shadowHost
+    prefix = '__next-dev-server-watcher-'
+  }
+
+  // Container
+  const container = createContainer(prefix)
+  shadowRoot.appendChild(container)
+
+  // CSS
+  const css = createCss(prefix)
+  shadowRoot.appendChild(css)
+  const closeEl = container.querySelector(`#${prefix}close`)
+
+  // State
+  let showBanner = false
+  const dismissKey = '__NEXT_DISMISS_DEV_SERVER_INDICATOR'
+  const dismissUntil = parseInt(window.localStorage.getItem(dismissKey), 10)
+  const dismissed = dismissUntil > new Date().getTime()
+
+  // Handle events
+  const evtSource = getEventSourceWrapper({ path: '/_next/webpack-hmr' })
+
+  evtSource.addConnectionListener((online) => {
+    showBanner = !online
+    updateContainer()
+  })
+
+  closeEl.addEventListener('click', () => {
+    const oneHourAway = new Date().getTime() + 1 * 60 * 60 * 1000
+    window.localStorage.setItem(dismissKey, oneHourAway + '')
+    showBanner = false
+    updateContainer()
+  })
+
+  function updateContainer() {
+    if (showBanner && !dismissed) {
+      container.classList.add(`${prefix}visible`)
+    } else {
+      container.classList.remove(`${prefix}visible`)
+    }
+  }
+}
+
+function createContainer(prefix) {
+  const container = document.createElement('div')
+  container.id = `${prefix}container`
+  container.innerHTML = `
+    <div id="${prefix}wrapper">
+      <h1>Development server disconnected</h1>
+      <button type="button" id="${prefix}close" title="Hide indicator for session">
+        <span>×</span>
+      </button>
+    </div>
+  `
+
+  return container
+}
+
+function createCss(prefix) {
+  const css = document.createElement('style')
+  css.textContent = `
+    #${prefix}container {
+      position: absolute;
+      top: 0;
+      width: 100%;
+
+      border-radius: 3px;
+      background: #f55;
+      color: #fff;
```

Comment:
This contrast ratio is quite low, only 3.1, which is a fail for WCAG AAA. See the results here: https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=FF5555

Perhaps you could instead use a lighter background colour, with a darker text colour, like this: https://webaim.org/resources/contrastchecker/?fcolor=9B1C1C&bcolor=FDE8E8?